trimesh/util.py

```diff
@@ -65,7 +65,7 @@ def unitize(points, check_valid=False):
         length[np.isnan(length)] = 0.0
 
     if check_valid:
-        valid = np.greater(length, _TOL_ZERO)
+        valid = np.not_equal(length, 0.0)
```

Comment:
This change is interesting and I think for the best -- we might want to just trust face normals unless they're actually zero.

Reply:
Since these are floats, shouldn't it really be `np.logical_not(np.isclose(length, 0.0))`

Reply:
But yeah 👍 on using numpy compare functions

Reply:
Oh right, excellent point. I'll update that now.

Reply:
Actually, looks like behavior should be the same assuming you want rtol = atol = 0.0 in the np.isclose() call. That is, assuming you want to only mark normals that are actually zero-length as invalid and are okay with very small faces.

Reply:
I think we want a tolerance on the float comparison, comparing to exactly 0.0 seems likely to be squirrely. The real question is how small to go- currently tol.merge connects vertices roughly closer than 1e-8. So if we have a face with two 1e-8 edges the magnitude of the cross product would be 1e-16. What about a comparison ~= that?

Reply:
That sounds like a great idea! The cross product magnitude also has a sin(theta) term though, so we'll need to incorporate allowable angles somehow -- if you allow angles as small as 1e-12 rad, suddenly the floor becomes 1e-28. What do you think a reasonable threshold on the angle is?

Reply:
Ah good point on the sin term I forgot about. So `tol.zero` in trimesh is 1e12, and looks like [machine epsilon](https://en.wikipedia.org/wiki/Machine_epsilon) for float64 is ~1e-16. So I should probably make that normal magnitude check some number larger than machine epsilon- 1e-15?

Reply:
Numpy has a nice function for machine epsilon actually, might just make everything use tol.zero:
`tol.zero = np.finfo(np.float64).resolution * 100`